tools/InterfaceGenerator/generator/generators/SmartFactoryBase.py

```diff
@@ -840,7 +840,28 @@ def _gen_schema_item_decl_code(self, param, member_name, default_value):
                     [[u"bool", None if param.default_value is None
                       else u"true" if param.default_value is True else u"false"]]))
         elif type(param) is Integer:
-            if not param.max_value or param.max_value and param.max_value < 2 ** 31:
+            if param.min_value is None and param.max_value is None:
```

Comment:
@RomanReznichenkoLuxoft `int64_t` should be generated not only in this case. It should be generated when:
- Both man and max are null
- Min is null
- Min less than min(int32_t)
- Max is null or >= max(int32_t) and min < 0